projects/vdk-core/src/vdk/internal/builtin_plugins/version/new_version_check_plugin.py

```diff
@@ -43,48 +43,68 @@ def vdk_configure(self, config_builder: ConfigurationBuilder) -> None:
         except Exception as e:
             log.info(
                 f"Could not check for new version release. "
                 f"Error was {e}. We are ignoring the error."
             )
 
     @staticmethod
-    def check_version(package_name, package_index, is_plugin=False):
-        if new_package(package_index, package_name).check():
-            # if package index is not specified that we fetch it from pip repo (pypi.org)
-            extra_index_if_needed = (
-                f"--extra-index-url {package_index}" if package_index else ""
-            )
-            message = (
-                f"New version for a plugin {package_name} is Available"
-                if is_plugin
-                else f"New Version of {package_name} is Available"
-            )
-            # We are using eager strategy so that if there's new version of a plugin that satisfies the requirements
-            # of the main package we want to upgrade to it.
-            pip_command = "pip install --upgrade-strategy eager -U"
-            click.echo(
-                f"""
-                **********************************************************************************************************************
+    def check_version(package_list, package_index):
+        not_single_package = len(package_list) > 1
 
-                                                {message}
+        # if package index is not specified that we fetch it from pip repo (pypi.org)
+        extra_index_if_needed = (
+            f"--extra-index-url {package_index}"
+            if (package_index and (package_index != "https://pypi.org"))
+            else ""
+        )
+        pip_command = "pip install --upgrade-strategy eager -U"
+        main_message = f"New version{'s' if not_single_package else ''} for {', '.join(package_list)} {'are' if not_single_package else 'is'} available."
+        install_message = (
+            f"{pip_command} {' '.join(package_list)} {extra_index_if_needed}"
+        )
+        # We are using eager strategy so that if there's new version of a plugin that satisfies the requirements
+        # of the main package we want to upgrade to it.
+        click.echo(
+            f"""
+    ******************************************************************************************
 
-                  Please update to latest version by using:
-                  {pip_command} {package_name} {extra_index_if_needed}
+    {chunk_message(main_message, 80)}
 
-                **********************************************************************************************************************
-                """,
-                err=True,
-            )
+    Please update to latest version by using:
+    {install_message}
+
+    ******************************************************************************************
+            """,
+            err=True,
+        )
+
+
+def chunk_message(message, line_len):
```

Comment:
Make private or protected . Also write pydoc.  

But is it necessary. Wouldn't https://docs.python.org/3/library/textwrap.html work ?

Reply:
It would. I knew there should be a library for this but i couldn't figure out how to find it, so I settled on rewriting it.